File: framework/web/src/test/java/top/bettercode/simpleframework/test/TestController.java
package top.bettercode.simpleframework.test;


import java.io.Serializable;
import java.util.Date;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import top.bettercode.lang.util.StringUtil;
import top.bettercode.simpleframework.support.code.ICodeService;
import top.bettercode.simpleframework.web.BaseController;
import top.bettercode.simpleframework.web.DataDicBean;
import top.bettercode.simpleframework.web.resolver.Cent;
import top.bettercode.simpleframework.web.resolver.CentConverter;
import top.bettercode.simpleframework.web.validator.ChinaCell;

/**
 * @author Peter Wu
 */
@SpringBootApplication
@RestController
@Validated

public class TestController extends BaseController {

  @RequestMapping(value = "/test")
  public Object test(DataDicBean form,@Cent Long cent, Date a,@ChinaCell String cell) {
    System.err.println(a);
    System.err.println(cent);
    System.err.println(form.getPrice());
    System.err.println(StringUtil.valueOf(form, true));
    DataDicBean dataDicBean = new DataDicBean();
    dataDicBean.setCode("code");
    dataDicBean.setIntCode(1);
    dataDicBean.setPath("/abc.jpg");
    return ok(dataDicBean);
  }

  @RequestMapping(value = "/errors")
  public Object error() {
    throw new RuntimeException("xx");
  }

  @Configuration(proxyBeanMethods = false)
  protected static class CodeConfiguration {

    @Bean
    public CentConverter yuanToCentConverter() {
      return new CentConverter();
    }

    @Bean
    public ICodeService codeNumberService() {
      return new ICodeService() {
        @Override
        public String getName(String codeType, Serializable code) {
          return "name";
        }

        @Override
        public Number getCode(String codeType, String name) {
          return 1;
        }
      };
    }


  }
}

File: util/api-sign/src/test/kotlin/top/bettercode/api/sign/TestController.kt
package top.bettercode.api.sign

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.servlet.http.HttpServletRequest

/**
 * @author Peter Wu
 * @since 1.1.4
 */
@SpringBootApplication
@RestController
class TestController {

    @RequestMapping("/test")
    fun test(httpServletRequest: HttpServletRequest): String {
        httpServletRequest.parameterMap.forEach { (t, u) ->
            System.err.println("$t:${u.joinToString()}")
        }
        return "success"
    }

}
@SpringBootApplication
@RestController
@ApiSignIgnore
class ApiSignIgnoreTestController {

    @RequestMapping("/apiSignIgnore/type")
    fun test(): String {
        return "success"
    }

}
@SpringBootApplication
@RestController
class ApiSignIgnoreMethodController {

    @ApiSignIgnore
    @RequestMapping("/apiSignIgnore/method")
    fun test(): String {
        return "success"
    }

}
